Add tab color indicator for tab switch menu (CTRL+T)

nukoseer opened the pull request:
Added tab color indicator for the tab switch menu. Tab color indicators have the same color as the background color of the tabs. If a tab has the default background color, the indicator is not shown in the tab switch menu. 

_I don't know if it was a good idea, but I just used small circles for indicators._

Relevant issue(s); #17465 

https://github.com/user-attachments/assets/a183031b-24cb-4349-89e1-6df9382cf9ad

lhecker reviewed (approved):
Neat. 😺

zadjii-msft commented on src/cascadia/TerminalApp/CommandPalette.xaml:
```diff
@@ -205,6 +205,13 @@
 
                     </StackPanel>
+
+                    <Ellipse Grid.Column="4"
```
Slightly worried that this will overlap with the scrollbar on the tab picker if there's a _lot_ of tabs. Who knows - maybe that column was vestigial?

nukoseer replied:
It kind of overlaps, but it doesn't seem like a big problem to me 😅 

![resim](https://github.com/user-attachments/assets/21e7dbfa-e9d0-40cc-b144-3cacf322ed8c)

zadjii-msft commented on src/cascadia/TerminalApp/TabBase.cpp:
```diff
@@ -340,6 +340,22 @@ namespace winrt::TerminalApp::implementation
+    {
+        const auto currentColor = GetTabColor();
+
+        if (currentColor.has_value())
```
IIRC this if/else could probably just be a single `.value_or(Windows::UI::Colors::Transparent())`

zadjii-msft reviewed (approved):
Yea, these are just nits. This is neat, thanks!

DHowett commented:
I love this so much! Thanks for doing it! 😄

DHowett reviewed (changes requested):
Hey, I did just realize that we have `TabPaletteItem` specifically for things like this. You can see where it's used by finding its members `IsReadOnlyActive` and `IsInputBroadcastActive`.

I was feeling a little weird about it being on the generic `PaletteItem`, because that's also used for commands. I'm thinking that it might feel better on `TabPaletteItem`!

nukoseer commented:
> Hey, I did just realize that we have `TabPaletteItem` specifically for things like this. You can see where it's used by finding its members `IsReadOnlyActive` and `IsInputBroadcastActive`.

I removed `TabColorIndicator` from `PaletteItem` and moved it to `TerminalTabStatus`. I also moved the code from `TabBase.cpp` to `TerminalTab.cpp`. Sorry for the confusion. I hope it looks better now.

DHowett reviewed (approved):
Beautiful! Only +13 lines, wow
